packages/datagateway-dataview/src/views/table/isis/isisStudiesTable.component.tsx

```diff
@@ -114,28 +114,28 @@ const ISISStudiesTable = (
       onSort={pushSort}
       columns={[
         {
-          icon: <TitleIcon />,
+          icon: <FingerprintIcon />,
           label: t('studies.name'),
           dataKey: 'STUDY.NAME',
           cellContentRenderer: (props: TableCellProps) =>
             tableLink(
-              `/${pathRoot}/instrument/${instrumentId}/${instrumentChild}/${props.rowData.ID}/investigation`,
+              `/${pathRoot}/instrument/${instrumentId}/${instrumentChild}/${props.rowData.STUDY.ID}/investigation`,
```

Comment:
When I load the table and click on the first instrument, when loading the study table I now get an error: `props.rowData.STUDY is undefined` which I presume stems from here. However,  if I load the page directly (e.g. reloading the page once I get the error) it works fine. This error only occurs in Table view, but if I navigate via Card view then switch to Table view then the data loads correctly with no error. It seems as if the data isn't being requested when the route changes?

I think it might be because of the two useEffects required for fetching data in this table.

Reply:
When navigating directly to studies table `data` is empty, and when swapping from cardview to table we already have `data` populated with `StudyInvestigation`s. In both cases `props.rowData.STUDY` can be called safely.

When navigating from instruments `data` is populated with `Instrument`s which obviously don't have `STUDY` on them, and so it errors (`clearTable` is called straight away but the error still occurs before it can reset `data`).

We don't see the issue with card view as due to the pagination logic, the link isn't rendered straight away, only after we've called `fetchData` once. All other tables use `<entity>.ID` so this isn't an issue normally, as the new table can still access an ID from the old `data` for however long the split second is before `clearTable` hits.

Rather than fundamentally change anything to do with routing or clearing, just changing it to `STUDY?.ID` will ensure we don't error while waiting between the location change and `clearTable`.